overlord/snapstate/target.go

```diff
@@ -677,15 +704,53 @@ func (p *pathInstallGoal) toInstall(ctx context.Context, st *state.State, opts O
 		return nil, err
 	}
 
+	comps, err := componentSetupsFromPaths(info, p.components)
+	if err != nil {
+		return nil, err
+	}
+
 	inst := target{
 		setup: SnapSetup{
 			SnapPath:  p.path,
 			Channel:   channel,
 			CohortKey: p.revOpts.CohortKey,
 		},
-		info:   info,
-		snapst: snapst,
+		info:       info,
+		snapst:     snapst,
+		components: comps,
 	}
 
 	return []target{inst}, nil
 }
+
+func componentSetupsFromPaths(snapInfo *snap.Info, components map[*snap.ComponentSideInfo]string) ([]ComponentSetup, error) {
+	setups := make([]ComponentSetup, 0, len(components))
+	for csi, path := range components {
+		compInfo, err := validatedComponentInfo(path, snapInfo, csi)
```

Comment:
thank you, wondering did we miss any validation on the store flows?

Reply:
Most of the validation happens in [componentSetupFromResource](https://github.com/snapcore/snapd/blob/master/overlord/snapstate/target.go?plain=1#L331). We check that the component from the resource exists in the `snap.yaml` that we get back from the store, and we check that the resource's type matches the type that is defined in the `snap.yaml`. I think that is really all we can do at that point?

And then there is the stuff that happens in `doValidateComponent` with assertions, of course.